### PR TITLE
Chore/#182201658 setup project environment

### PR DESCRIPTION
**Pull request by mwisemarierose**
#### What does this PR do?
this PR is created so that all folder that will be needed in project  are visible

#### Any background context you want to provide?
In this PR there is a `global handling Error` function which will display all the error that are captured in the stack trace

Screenshot (5)

We also made it simple for everyone to send `Error message` with the `status code`  function which will help the team to write related code base.

-first of all you need to import the `AppError.js` file in order to use that function

Screenshot (4)

-the function needs to be wrapped in `next` function so that it can send it in error capture stack trace, it also takes 2 arguments `message` and `status code`

Screenshot (3)

- **with this changes no one is allowed to define any endpoint in `index.js` or `app.js` you will manage to define them in `src/router/index.js`**
- **also we decided to put everything that is related to the server in `index.js` and everything that is related to express in `app.js`**

**_The main reason  for all this changes is reduce the work of code rebasing_.**

#### What are the relevant pivotal tracker stories?
Set up an empty Express Boilerplate with dotenv #182201658

### CODE REVIEW

**Review by Placide123 — DISMISSED** [review]

**Review by alaintresor — DISMISSED** [review]

**Review by joshwambere — CHANGES_REQUESTED** [review]
@mwisemarierose  controllers folder is missing the same applies to routes and test folders

**Review by semy-tuyi — APPROVED** [review]

**Review by semy-tuyi — DISMISSED** [review]

**Review by uwambajeddy — DISMISSED** [review]
LGTM

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,4 @@ dist
 
 # TernJS port file
 .tern-port
+.husky
```
Comment (didasmbalanya) [inline .gitignore]: Do not ignore the .husky file. its important

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import morgan from 'morgan';
+import '@babel/polyfill';
```
Comment (didasmbalanya) [inline src/app.js]: move to first line of the file

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+const app = express();
+
+app.use(cors());
+app.use(morgan('dev'));
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+app.use(express.static(`${__dirname}/public`));
```
Comment (didasmbalanya) [inline src/app.js]: remove, we will not be serving any static files for now

**Review by didasmbalanya — CHANGES_REQUESTED** [review]

**Review by uwambajeddy — DISMISSED** [review]
LGTM

File: new.json [hunk 4]
```diff
@@ -0,0 +1,88 @@
+{
```
Comment (joshwambere) [inline new.json]: @mwisemarierose what is the purpose of this file

**Review by joshwambere — DISMISSED** [review]

**Review by uwambajeddy — DISMISSED** [review]

**Review by joshwambere — APPROVED** [review]

**Pull request merged**